examples/some-issuer/src/lib.rs

```diff
@@ -0,0 +1,176 @@
+use std::collections::BTreeMap;
+use std::sync::Arc;
+
+use anyhow::Context;
+use axum::Json;
+use axum_sessions::async_session::chrono::{self, TimeZone};
+use concordium_rust_sdk::cis4::{Cis4Contract, Cis4TransactionMetadata};
+use concordium_rust_sdk::common::types::{KeyPair, TransactionTime};
+use concordium_rust_sdk::contract_client::CredentialInfo;
+use concordium_rust_sdk::id::constants::{ArCurve, AttributeKind};
+use concordium_rust_sdk::id::pedersen_commitment;
+use concordium_rust_sdk::smart_contracts::common::{Amount, Duration, Timestamp};
+use concordium_rust_sdk::types::hashes::TransactionHash;
+use concordium_rust_sdk::types::transactions::send::GivenEnergy;
+use concordium_rust_sdk::types::{CryptographicParameters, Energy, Nonce, WalletAccount};
+use concordium_rust_sdk::web3id::did::Network;
+use concordium_rust_sdk::web3id::{SignedCommitments, Web3IdAttribute, Web3IdCredential};
+use reqwest::{StatusCode, Url};
+use serde::Serialize;
+
+#[derive(Clone)]
+pub struct IssuerState {
+    pub crypto_params: Arc<CryptographicParameters>,
+    pub contract_client: Cis4Contract,
+    pub network: Network,
+    pub issuer: Arc<WalletAccount>,
+    pub issuer_key: Arc<KeyPair>,
+    pub nonce_counter: Arc<tokio::sync::Mutex<Nonce>>,
+    pub max_register_energy: Energy,
+    pub metadata_url: Arc<Url>,
+    pub credential_schema_url: Arc<Url>,
+}
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct IssueResponse {
+    tx_hash: TransactionHash,
+    credential: Web3IdCredential<ArCurve, Web3IdAttribute>,
+}
+
+#[tracing::instrument(level = "info", skip(issuer, credential, user_id))]
+pub async fn issue_credential(
+    issuer: IssuerState,
+    credential: CredentialInfo,
+    user_id: String,
+) -> Result<Json<IssueResponse>, StatusCode> {
+    tracing::info!("Request to issue a credential.");
+
+    if let Err(err) = validate_credential(&issuer, &credential) {
+        tracing::warn!("Failed to validate credential: {err}");
+        return Err(StatusCode::BAD_REQUEST);
+    }
+
+    match register_credential(issuer, credential, user_id).await {
+        Ok(res) => {
+            tracing::info!("Successfully issued credential.");
+            Ok(Json(res))
+        }
+        Err(err) => {
+            tracing::error!("Failed to register credential: {err}");
+            Err(StatusCode::INTERNAL_SERVER_ERROR)
+        }
+    }
+}
+
+/// Checks that the credential is reasonable.
+fn validate_credential(issuer: &IssuerState, credential: &CredentialInfo) -> anyhow::Result<()> {
+    anyhow::ensure!(
+        credential.holder_revocable,
+        "Credential should be holder revocable."
+    );
+    let now = chrono::Utc::now().timestamp_millis();
+    let delta =
+        Timestamp::from_timestamp_millis(now as u64).duration_between(credential.valid_from);
```

Comment:
```suggestion
    let now = chrono::Utc::now().timestamp_millis();
    let delta = now.signed_duration_since(credential.valid_from);
```

Reply:
The suggestion doesn't work (I assume you also meant to remove `.timestamp_millis()`). `credential.valid_from` is of type `concordium_contracts_common::Timestamp`.